proposals/0006-react-native-monorepo.md

```diff
@@ -0,0 +1,229 @@
+---
+title: React Native as a monorepo
+author:
+  - Lorenzo Sciandra
+  - Tommy Nguyen
+  - Nicola Corti
+date: 2022-05-18
+---
+
+# RFC0006: React Native as a monorepo
+
+## Summary
+
+This proposal is not much as in introducing something completely new, but more to discuss a solution to an existing problem. To put simply, the current shape of the `react-native` codebase hosted in GitHub is lacking structure and consistency and this is leading to a number of issues (detailed below).
+
+This proposal is about re-shaping the existing codebase on GitHub into what is commonly expected of a monorepo in the JavaScript ecosystem (see, for example, [Babel.js](https://github.com/babel/babel) or [Next.js](https://github.com/vercel/next.js)): consistent versioning, naming and tooling.
+
+### Table of Content
+
+- [Motivation](#motivation)
+  - [A IRL scenario of this being a problem: `react-native-codegen` and RN 68](#a-irl-scenario-of-this-being-a-problem-react-native-codegen-and-rn-68)
+- [Detailed design](#detailed-design)
+  - [The proposal](#the-proposal)
+  - [Going the extra mile: Hermes](#going-the-extra-mile-hermes)
+  - [Implementation detail considerations](#implementation-detail-considerations)
+  - [Other considerations](#other-considerations)
+- [Plan of action](#plan-of-action)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+- [Future work](#future-work)
+- [Unresolved questions](#unresolved-questions)
+
+## Motivation
+
+To quickly explain the motivation behind this proposal, here's the current shape of the React Native codebase hosted on GitHub and corrispective npm packages:
+
+![current status of the react-native GitHub codebase](/proposals/assets/0006-current-status.png "Current Status")
```

Comment:
Can we use a relative URL for both images? Not displaying in GitHub preview and unclear if this will work in other previewers.

Reply:
I'm not sure which preview you are referring to, but when I look it here https://github.com/kelset/discussions-and-proposals/blob/kelset/proposal-react-native-monorepo/proposals/0006-react-native-monorepo.md they should up?

Also, I guess you mean absolute urls? ex. https://github.com/kelset/discussions-and-proposals/blob/kelset/proposal-react-native-monorepo/proposals/assets/0006-current-status.png ?

Reply:
Fair! Where I saw this not working was in the [PR files preview](https://github.com/react-native-community/discussions-and-proposals/pull/480/files/4ea9fb3e93e3a814812d406551e07dd7d4b3fe4a?short_path=d97c7fa#diff-d97c7fabe0f53f5637e08b32c195b4fc945add1cf98ebeb34ec2279f9e577cbc). I'd expect that `![](./assets/0006-current-status.png)` might work in more places, such as local text editors also. Super minor point.

Reply:
ohhh I see, ok no problem let me address that - will tackle it tomorrow